onnxruntime/test/optimizer/qdq_transformer_test.cc

```diff
@@ -3725,19 +3727,127 @@
   };
 
   test_case({{1, 2, 4}, {1, 3, 4}}, false, false);  // Do not block removal
-  test_case({{1, 2, 4}, {1, 3, 4}}, true, false);   // Block removal of first dq
-  test_case({{1, 2, 4}, {1, 3, 4}}, false, true);   // Block removal of last dq
+  test_case({{1, 2, 4}, {1, 3, 4}}, true, false);   // Block removal of last dq
+  test_case({{1, 2, 4}, {1, 3, 4}}, false, true);   // Block removal of first dq
   test_case({{1, 2, 4}, {1, 3, 4}}, true, true);    // Block removal of first and last dq
 
 #if !defined(DISABLE_CONTRIB_OPS)
   // Use contrib QDQ ops
   test_case({{1, 2, 4}, {1, 3, 4}}, false, false, true);  // Do not block removal
-  test_case({{1, 2, 4}, {1, 3, 4}}, true, false, true);   // Block removal of first dq
-  test_case({{1, 2, 4}, {1, 3, 4}}, false, true, true);   // Block removal of last dq
+  test_case({{1, 2, 4}, {1, 3, 4}}, true, false, true);   // Block removal of last dq
+  test_case({{1, 2, 4}, {1, 3, 4}}, false, true, true);   // Block removal of first dq
   test_case({{1, 2, 4}, {1, 3, 4}}, true, true, true);    // Block removal of first and last dq
 #endif
 }
 
+// test removal of Q->DQs pairs by QDQFinalCleanupTransformer
+TEST(QDQTransformerTests, QDQFinalCleanupTransformer_BasicQDQsCleanup) {
+  auto test_case = [&](bool is_input_q,
```

Comment:
could you please also add a DQ->Qs test case?

Reply:
Added